codec/src/main/java/io/netty/handler/codec/AsciiString.java

```diff
@@ -656,7 +501,7 @@ public boolean equalsIgnoreCase(CharSequence string) {
         }
 
         for (int i = 0; i < thisLen; i++) {
-            char c1 = (char) (value[i] & 0xFF);
+            char c1 = (char) value[i];
```

Comment:
This is not correct.

    System.err.println((int)((char) ((byte) -1)));

Will print `65535`. What we want is `255`.